blog/_posts/2020-10-29-load-dynamic.md

```diff
@@ -0,0 +1,126 @@
+---
+slug: "load-dynamic"
+date: 2020-10-29
+author: nick
+layout: blog
+title: "Load Dynamic"
+subtitle: "Or: How Build Tools Grow"
+image: "/assets/images/load-dynamic/briggs_house.jpg"
+image_caption: "Chicago used to load their buildings at runtime. <a href='https://commons.wikimedia.org/wiki/File:Briggs_house.jpg'>Photo from the Chicago Historical Society via Wikipedia.</a>"
+tags:
+  - tilt
+  - bazel
+---
+
+If you have more than one service, you probably need to share some common
+functions and constants between services.
+
+The way we solve this in Tilt is with two primitives: `load()` and
+`load_dynamic()`. They have slightly different syntaxes.
+
+Load is static: it takes a literal string, and loads new variables into the
+current scope. Here's how you load a helper function `create_namespace`:
+
+```
+load('./lib/Tiltfile', 'create_namespace')
+create_namespace('frontend')
+```
+
+LoadDynamic is...well...dynamic! It's an imperative function that takes a string
+as input, and returns a dictionary of values:
+
+
+```
+values = load_dynamic('./lib/Tiltfile')
+create_namespace = values['create_namespace']
+create_namespace('frontend')
+```
+
+If you want more detail on how to use them, mosey on over to [the Tilt API
+reference](https://docs.tilt.dev/api.html).
+
+If you want to know why we have two ways to load files, read on!
+
+## Why do you need an imperative load_dynamic()?
```

Comment:
I feel like this post currently only really answers this question at an architecture astronaut level.

For a Tilt user who's interested in a more concrete example of how this affects their day-to-day, I think it'd be useful to change the `load_dynamic` example to something like:
```
lib = base_path + '/lib/Tiltfile'
# can load from variables, not just string literals!
values = load_dynamic(lib)
```

(or maybe the existing example demonstrates the solution to problems that aren't coming to my mind at the moment?)